wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift

```diff
@@ -113,26 +115,43 @@ final class ConversationCreationController: UIViewController {
 
     private lazy var encryptionProtocolSection = {
         let section = ConversationEncryptionProtocolSectionController(values: values)
-        section.isHidden = true
+
         section.tapAction = { sender in
             self.presentEncryptionProtocolPicker(sender: sender) { [weak self] encryptionProtocol in
-                self?.values.encryptionProtocol = encryptionProtocol
-                self?.updateOptions()
+                guard let self else { return }
+
+                self.values.encryptionProtocol = encryptionProtocol
+                self.updateOptions()
+
+                self.reloadOptionsSections()
             }
         }
         return section
     }()
 
+    private func reloadOptionsSections() {
+        guard let collectionView = collectionViewController.collectionView else { return }
+        updateSections()
+
+        let startIndex = 2
```

Comment:
``startIndex` is 2 because we don't want to reload the first 2 sections (nameSection and errorSection) when the message protocol changes, as this will clear the conversation name.`
I've added this comment to the code

Reply:
I think if you can it's better to calculate the indices rather than hardcode them. Hardcoded indices are brittle and can lead to out of bound exceptions. Another option is to modify the name section to take the current 'values' like the other sections, this way even if the section is reloaded it preserves the current name rather than starting with an empty name.

Reply:
I removed the hardcode value and changed the code. Please take a look